src/main/kotlin/org/jetbrains/research/testspark/helpers/psiHelpers/JavaPsiHelper.kt

```diff
@@ -2,121 +2,84 @@ package org.jetbrains.research.testspark.helpers.psiHelpers
 
-        return "${psiMethod.name}($parameterTypes)$returnType"
-    }
-
-    override fun getSurroundingClass(
-        psiFile: PsiFile,
-        caretOffset: Int,
-    ): PsiClass? {
-        // Get the classes of the PSI file
-        val classElements = PsiTreeUtil.findChildrenOfAnyType(psiFile, PsiClass::class.java)
-
-        // Get the surrounding PSI class (i.e. the cursor has to be within that class)
-        var surroundingClass: PsiClass? = null
-        for (psiClass: PsiClass in classElements) {
-            if (withinElement(psiClass, caretOffset)) {
-                // Check the constraints on a class
-                if (!validateClass(psiClass)) continue
-                surroundingClass = psiClass
-            }
+            return "${psiMethod.name}($parameterTypes)$returnType"
         }
-        return surroundingClass
-    }
-
-    override fun getSurroundingMethod(
-        psiFile: PsiFile,
-        caretOffset: Int,
-    ): PsiMethod? {
-        // Get the methods of the PSI file
-        val methodElements = PsiTreeUtil.findChildrenOfAnyType(psiFile, PsiMethod::class.java)
-
-        // Get the surrounding PSI method (i.e. the cursor has to be within that method)
-        var surroundingMethod: PsiMethod? = null
-        for (method: PsiMethod in methodElements) {
-            if (isMethodConcrete(method) && withinElement(method, caretOffset)) {
-                val surroundingClass: PsiClass =
-                    PsiTreeUtil.getParentOfType(method, PsiClass::class.java) ?: continue
-                // Check the constraints on the surrounding class
-                if (!validateClass(surroundingClass)) continue
-                surroundingMethod = method
-            }
+    override val signature: String
+        get() {
+            val bodyStart = psiMethod.body?.startOffsetInParent ?: psiMethod.textLength
+            return psiMethod.text.substring(0, bodyStart).replace('\n', ' ').trim()
         }
-        return surroundingMethod
-    }
 
-    override fun getSurroundingLine(
-        psiFile: PsiFile,
-        caretOffset: Int,
-    ): Int? {
-        val psiMethod: PsiMethod = getSurroundingMethod(psiFile, caretOffset) ?: return null
-
-        val doc: Document = PsiDocumentManager.getInstance(psiFile.project).getDocument(psiFile) ?: return null
+    override val text: String? = psiMethod.text
 
-        val selectedLine: Int = doc.getLineNumber(caretOffset)
-        val selectedLineText: String =
-            doc.getText(TextRange(doc.getLineStartOffset(selectedLine), doc.getLineEndOffset(selectedLine)))
+    override val containingClass: PsiClassWrapper? = psiMethod.containingClass?.let { JavaPsiClassWrapper(it) }
 
-        if (selectedLineText.isBlank()) return null
-
-        if (!validateLine(selectedLine, psiMethod, psiFile)) return null
-        return selectedLine
+    override val containingFile: PsiFile = psiMethod.containingFile
+    override fun isLineIn(lineNumber: Int): Boolean {
```

Comment:
**[neatpick]** I would rename this method to `containsLine`; in this case, a call of this method would be semantically a bit more natural `method.containsLine(10)` compared to `method.isLineIn(10)`.